### ISSUE TITLE
Event handlers will not run when performing write operations

### ISSUE THREAD

**Issue by deanebarker** [Shazwazza/UmbracoLinqPadDriver#4]
I do not think this is fixable, but I thought I'd note that event handlers wired up in the website code will _not_ run when you do write operations from LINQPad. This only makes sense, since those are in the web server context, which doesn't apply here.

Something to be aware of.

**Comment by Shazwazza** [comment]
Indeed, this is done on purpose. All non Umbraco Core event handlers are removed on booting otherwise we'll end up with most likely a ton of errors. Any web based concepts cannot be used on background threads which is what this library is doing. 

It could be possible (perhaps) to add some custom attribute to startup handlers that this driver is aware of and will not remove them but it would have to be either a custom attribute that this driver ships with or another shared .NET or Umbraco attribute that both your code and this driver is aware of. 

For example:

```cs
[BackgroundThreadSafe]
public class MyStartup : UmbracoEventHandler {

}
```

**Comment by Shazwazza** [comment]
see: https://github.com/Shazwazza/UmbracoLinqPadDriver/blob/master/UmbracoLinqPad.Gateway/Bootstrap/ConsoleBootManager.cs#L86

**Comment by deanebarker** [comment]
It would be nice to make this optional/configurable. I have some event handlers I had to replicate in LINQPad because they were removed. I think there's some value in allowing selected handlers to run.

**Comment by Shazwazza** [comment]
Just not sure how that would work? The only way would be to extend the dialog for the connection details to an Umbraco source allowing you to specify (by string) a list of event handlers that shouldn't be excluded.

**Comment by deanebarker** [comment]
I think we could do it in `appSettings` config, on either side: in the config of the LINQPad script, or in the web.config on the Umbraco site, since you copy those variables over to the LINQPad instance.

Which side is more appropriate?

**Comment by Shazwazza** [comment]
Sorry for the delay! There's not really a config file available from the linqpad side of thing that would be easy to use. I think we can do this in two ways:

1. special appSetting in the web.config on the umbraco side (easy)
1. custom dialog (maybe 'advanced settings') when setting up the linqpad provider connection to specify the list of items (harder)

We can do both of these things but 1) is easiest so can start there.

**Comment by deanebarker** [comment]
LINQPad config is pretty easy -- if you press F4, there's a tab for "App.config" and you can just type a config in there, without having to have an actual file. The config gets embedded and loaded directly from the `.linq` file.

But, I don't think it matters. Either/or, I guess.

**Comment by Shazwazza** [comment]
ah that's wonderful, ok well i'll see what i can whip up when i find some time over the next week
